(SKR 1.3) Assign DIAG pins for homing dir

pfillion42 opened the pull request:
### Requirements
SKR 1.3
Stepstick that use stallguard diag pin
sensorless homing with stallguard
(X,Y,Z)_HOME_DIR == 1

### Description


Added a condition to correctly assign the pins of SKR 1.3 when using sensorless homing in positive direction  with stallguard. In the case that the printer use positive homing direction, marlin use (X,Y,Z)_MAX_PIN to trigger but the stallguard diagnostic pin is physically on (X,Y,Z)_MIN_PIN


### Benefits

The code automatically validates the need to change the allocation of pins

### Related Issues

pfillion42 committed:
Update pins_BTT_SKR.h

To assign the pins and apply them to the right logical endstops. In the case where:

1.There is one or more TMC2130 steppers used on the Z axis
2.I use diagnostic pills (Stallgard sensorless homing) to activate endstops
3. The homing direction on the Z axis is upwards

As it is specific to this board, I believe that the change can be included

pfillion42 committed:
Revert "Update pins_BTT_SKR.h"

This reverts commit 58211dc128e39c44f7591dccc3b52a3df89c73ac.

pfillion42 committed:
Update pins_BTT_SKR.h

To assign the pins and apply them to the right logical endstops. In the case where:

1.There is one or more TMC2130 steppers used on the Z axis
2.I use diagnostic pills (Stallgard sensorless homing) to activate endstops
3. The homing direction on the Z axis is upwards

As it is specific to this board, I believe that the change can be included

AnHardt commented on Marlin/src/pins/lpc1768/pins_BTT_SKR.h:
```diff
@@ -35,8 +35,15 @@
+  // Use normal endstop pins
+  #define Z_MIN_PIN          P1_25
+  #define Z_MAX_PIN          P1_24
+#endif
```
Why Z-axis only?
Why only with stallguard?
Why "TMC2130" only?
Why only this board?
Why at all?

pfillion42 replied:
Why Z-axis only?
No

Why only with stallguard?
Yes

Why "TMC2130" only?
You're right, not necessarily only tmc2130 but all the stepstick that uses the pin diag on the SKR 1.3

Why only this board?
Because it is related to the way SKR1.3 is manufactured.
The Z Stepstic connector diagnostic pin uses P1_25, the same as Z min Endstop.

If you are trying to use Stallguard to homing the Z axis (x cary assembly) on the upper brackets, as for the prusa i3, the diagnostic pin of the Z Stepstick will not trigger Z_MAX_PIN as it should.

Why at all?
That's all why.
Look SKR 1.3 Board layout
![SKR1 3](https://user-images.githubusercontent.com/33739777/69426430-414b4d00-0cfb-11ea-9849-6f970d87d4fa.jpg)

pfillion42 replied:
For me it is only for the z axis but, I can also do the pin validation for x and y also in the case where someone uses stallguard and home in the opposite direction.

pfillion42 committed:
Update pins_BTT_SKR.h

pfillion42 reviewed (commented):
This is the new commit with all axis validation

pfillion42 commented:
My bad, i just see that my description was not transfert in the request

thisiskeithb commented:
`pins_BTT_SKR.h` is shared by both the SKR 1.1 and SKR 1.3, so if you only want this apply to the SKR 1.3, it should be moved to `pins_BTT_SKR_V1_3.h`.

pfillion42 committed:
Modification of BTT_SKR's pins files

New version with limit switches pins re-transferred to SKR 1.1 and SKR 1.3 map files that include proposed changes to support the stallguard in a positive direction

pfillion42 commented:
> 
> 
> `pins_BTT_SKR.h` is shared by both the SKR 1.1 and SKR 1.3, so if you only want this apply to the SKR 1.3, it should be moved to `pins_BTT_SKR_V1_3.h`.

Good point!
I made the changes in the 3 files BTT_SKR so that the pins are treated by board

thinkyhead committed:
Update pins_BTT_SKR_V1_3.h

thinkyhead commented:
> the stallguard diagnostic pin is physically on `(X,Y,Z)_MIN_PIN`

For boards that have this as a fixed behavior, I see that our only option is to switch the pin definition so that the endstop pin corresponds to the DIAG pin which is in use. Perhaps in the future we'll have a more general way to route the DIAG pin to the endstop function, considering the DIAG pin a more formal member of the endstop pin family.
